packages/dds/tree/src/core/tree/visitDelta.ts

```diff
@@ -287,17 +287,28 @@ export interface DeltaVisitor {
 	 * @param source - The bounds of the range of nodes to detach.
 	 * @param destination - The key for a new detached field.
 	 * A field with this key must not already exist.
+	 * @param destinationDetachedNodeId - The first detached node id for the detached nodes.
```

Comment:
Nit: the name `destinationDetachedNodeId` implies the ID is associated with the destination when it is in fact associated with the node. That's the purpose of this ID system: you don't have to know where the node is to refer to it.

I propose:
```suggestion
	 * @param id - The ID assigned to the first detached node as a result of the detach. The other nodes in the detached range are assigned subsequent IDs.
```